src/foam/mlang/mlang.js

```diff
@@ -1449,6 +1449,12 @@ foam.CLASS({
 
   requires: [ 'foam.mlang.Constant' ],
 
+  javaImports: [
```

Comment:
Personally I much prefer writing fully qualified names in the java code for three reasons.

1) If we're writing so much java code that using javaimports saves a lot of time, then we might be writing too much code.

2) If we use fully qualified names in the javaCode itself, then its easy to cut and paste that code if we need to move it around, as the code is more self contained.

3) javaImports quickly gets out of date, if we remove some code its extra mental overhead to read the rest of the class and see if we are still using all the imports.  As a result the list tends to grow and never shrink.

But like I said, this is just a preference not (yet) a rule.  So fix it if you agree, or leave it if you don't agree.  At a later date we could debate and decide which one is truly better :)

Reply:
1. I don't understand what you mean by this
2. Fair enough
3. Fair enough

The reason I prefer to not use the full name is because I find it makes the code harder to read. Personally, to me it's visual noise, especially in an already verbose language like Java. The package name just adds superfluous text that the reader has to scan (and almost always ignore). The noise is especially annoying in nested code like mlangs where everything is prefixed by `foam.mlang.MLang.`. The only time I personally opt to add the full path is when there's a naming conflict.

I'd argue that code is read far more often than it's modified, and that's why I'd argue that it's more valuable to optimize for readability than optimizing for copy-and-pasting and modification.

Reply:
1 is kind of a pre-answer to a common argument.  Adding javaImports saves us typing because we don't have to type the fully qualified name every time we refer to something in the method body.  So many people would argue that its worth having becuase it saves typing.  My counter-argument to this point, is that if we're writing so much javaCode by hand that having javaImports is saving us significant amounts of effort, then that is actually a symptom of a deeper problem.  Namely that we're probably missing the chance to model something at a higher level and we should address _THAT_ issue rather than trying to optimize the rate at which we write java code.

But that's not the argument you're making so its less relevant.

Readability is an interesting point.  I agree that code is read more often than it is written.  I think I feel that the code is just as readable if not more readable with fully qualified names.  Often I care if its the Jetty HttpServer or the nanos one.  Or its the foam.nanos.auth.User vs the net.nanopay.auth.User, having the fully qualified name there means that I don't have to refer to a different part of the file to determine which one being referenced.

But I agree that in some cases its just noise that you have to skip over.  I'll have to mull on this further, since I'm now not sure which I prefer.  Interesting discussion, let's move ahead with the code as is.